workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/IReader.cs

```diff
@@ -7,13 +7,14 @@ namespace Improbable.Gdk.Core.MonoBehaviours
     ///     The Reader interface is used by MonoBehaviours to query information about a SpatialOS component and register
     ///     callbacks for updates, events and commands for component instances.
     /// </summary>
-    /// <typeparam name="TComponent">The data type for the SpatialOS component.</typeparam>
-    public interface IReader<TComponent>
-        where TComponent : ISpatialComponentData, IComponentData
+    /// <typeparam name="TSpatialComponentData">The data type for the SpatialOS component.</typeparam>
+    public interface IReader<TSpatialComponentData>
+        where TSpatialComponentData : ISpatialComponentData
     {
         Authority Authority { get; }
+        TSpatialComponentData Data { get; }
```

Comment:
Would users of specific subtypes see this property giving a general TSpatialComponentData in the API?